File: Travel Package management C++/TravelPackage.cpp
#include "TravelPackage.h"
#include "Destination.h"
#include "Passenger.h"
#include <iostream>


TravelPackage::TravelPackage(string n, int cap){
    name = n;
    capacity = cap;
}

void TravelPackage::addDestination(shared_ptr<Destination> destination){
    itinerary.push_back(destination);
}

void TravelPackage::addPassenger(shared_ptr<Passenger> passenger){
    passengerList.push_back(passenger);
}

void TravelPackage::printItinerary() {
    cout << "Travel Package Name: " << name << endl;
    cout << "Destinations:" << endl;
    for (const auto& destination : itinerary) {
        destination->printDetails();
        cout << endl; 
    }
}

void TravelPackage::printPassengerList() {
    cout << "Travel Package Name: " << name << endl;
    cout << "Passenger Capacity: " << capacity << endl;
    cout << "Passengers currently enrolled: " << passengerList.size() << endl;
    for(const auto& passenger : passengerList){
        cout << "Passenger Name: " << passenger->getName() << ", Passenger Number: " << passenger->getPassengerNumber();
    }
}

void TravelPackage::printAvailableActivities(){
    for(const auto& destination : itinerary){
        destination->printAvailableActivities();
    }
}

vector<shared_ptr<Destination>> TravelPackage::getItinerary(){ return itinerary; }
vector<shared_ptr<Passenger>> TravelPackage::getPassengerList(){ return passengerList; }

File: Travel Package Management Java/src/com/example/travel/TravelPackage.java
package com.example.travel;

import java.util.ArrayList;
import java.util.List;

public class TravelPackage {
    private String name;
    private int capacity;
    private List<Destination> itinerary = new ArrayList<>();
    private List<Passenger> passengerList = new ArrayList<>();

    public TravelPackage(String name, int capacity) {
        this.name = name;
        this.capacity = capacity;
    }

    public void addDestination(Destination destination) {
        itinerary.add(destination);
    }

    public void addPassenger(Passenger passenger) {
        if (passengerList.size() < capacity) {
            passengerList.add(passenger);
        } else {
            System.out.println("Cannot add more passengers, capacity reached.");
        }
    }

    public void printItinerary() {
        System.out.println("Travel Package Name: " + name);
        System.out.println("Destinations:");
        for (Destination destination : itinerary) {
            destination.printDetails();
            System.out.println();
        }
    }

    public void printPassengerList() {
        System.out.println("Travel Package Name: " + name);
        System.out.println("Passenger Capacity: " + capacity);
        System.out.println("Passengers currently enrolled: " + passengerList.size());
        for (Passenger passenger : passengerList) {
            System.out.println("Passenger Name: " + passenger.getName() + ", Passenger Number: " + passenger.getPassengerNumber());
        }
    }

    public void printAvailableActivities() {
        for (Destination destination : itinerary) {
            destination.printAvailableActivities();
        }
    }

    public List<Destination> getItinerary() {
        return itinerary;
    }

    public List<Passenger> getPassengerList() {
        return passengerList;
    }
}
